src/clients/fcoe-client_finish.rb

```diff
@@ -87,40 +87,35 @@ def main
           Builtins.y2milestone("Nothing to do")
         end
 
-        Builtins.foreach(@netcards) do |card|
-          command = ""
-          file_name = ""
-          if Ops.get_string(card, "fcoe_vlan", "") != FcoeClient.NOT_AVAILABLE &&
-              Ops.get_string(card, "fcoe_vlan", "") != FcoeClient.NOT_CONFIGURED
-            # FCoE VLAN interface is configured -> start services
-            @start_services = true
-
-            # copy sysconfig files
-            file_name = Builtins.sformat(
-              "/etc/sysconfig/network/ifcfg-%1.%2",
-              Ops.get_string(card, "dev_name", ""),
-              Ops.get_string(card, "vlan_interface", "")
-            )
-            command = Builtins.sformat(
-              "/usr/bin/cp -a %1 %2/etc/sysconfig/network",
-              file_name.shellescape,
-              Installation.destdir.shellescape
-            )
-            Builtins.y2milestone("Executing command: %1", command)
-            WFM.Execute(path(".local.bash"), command)
-
-            file_name = Builtins.sformat(
-              "/etc/sysconfig/network/ifcfg-%1",
-              Ops.get_string(card, "dev_name", "")
-            )
-            command = Builtins.sformat(
-              "/usr/bin/cp -a %1 %2/etc/sysconfig/network",
-              file_name.shellescape,
-              Installation.destdir.shellescape
-            )
-            Builtins.y2milestone("Executing command: %1", command)
-            WFM.Execute(path(".local.bash"), command)
-          end
+        @netcards.each do |card|
+          iface_name = FcoeClient.fcoe_vlan(card)
+          next if iface_name.nil?
+
+          # FCoE VLAN interface is configured -> start services
+          @start_services = true
+
+          # copy sysconfig files
+          file_name = "/etc/sysconfig/network/ifcfg-#{iface_name}"
+
+          command = Builtins.sformat(
```

Comment:
NP: When killing YCP zombies I would also try to kill all of them :P and even start using Yast::Execute :P

Reply:
Although in that case was probably easier to rewrite the client completely .. just fixing the issue should be enough by now.

Reply:
Not in release candidate phase and in a client that is not unit-testable without moving it first to a new proper class.